sys/dev/puc/puc_pci.c

```diff
@@ -198,3 +198,4 @@ static driver_t puc_pci_driver = {
 };
 
 DRIVER_MODULE(puc, pci, puc_pci_driver, puc_devclass, 0, 0);
+MODULE_PNP_INFO("U16:vendor;U16:device;V16:subvendor;V16:subdevice;D:#", pci, puc, puc_pci_devices, sizeof(puc_pci_devices[0]), nitems(puc_pci_devices) - 1);
```

Comment:
This change fails to adapt the puc_pci_devices table as described.  Additionally, long lines should be wrapped — see style(9).

Reply:
I'd  be more inclined to, in this case, leave out the subvendor and subdevice. They aren't needed to differentiate which cards the driver might load for, just to configure parameters of those cards. Therefore, it's easier to just ignore them (use U16:#; for each of the fields to ignore). That also solves the style issue.

Reply:
Actually cem, I think we should just ignore them. They don't change which driver gets the card, but do change how things are configured. I'd rather under-specify a smidge here than do the churn on the table.
"U16:vendor;U16:device;U16:#;U16:#;D:#;" seems like a much better choice for the descriptor string in this case.

Reply:
That's a good point.  Matching on vendor+devid is probably good enough -- even if it is false positive and some other driver is needed for the specific subvendor/subdevice, the harm is low -- one extra driver is loaded.

Should they be anonymous '#'s, or just remove them (and 'D:#') from the format string entirely?

Reply:
I think they should be anonymous #'s because I like to keep the description if possible. I'm still on the fence on its usefulness, and until I jump one way or the other, I'd like to tag as much as possible.

Reply:
I placed the anonymous #'s for now. For subvendor and subdevice, is it right to use U16 instead of V16 (which cemeyer has suggested) ? 

The puc_pci_devices array is a puc_cfg struct defined as extern const struct puc_cfg puc_pci_devices[];

The puc_cfg struct is
struct puc_cfg {
uint16_t vendor;
uint16_t device;
uint16_t subvendor;
uint16_t subdevice;
const char desc;
int clock;
int8_t ports;
int8_t rid; / Rid of first port /
int8_t d_rid; / Delta rid of next ports /
int8_t d_ofs; / Delta offset of next ports */
puc_config_f *config_function;
};

How can i adapt this PNP_INFO to puc_pci_devices table ?

Reply:
The PNP_INFO need only describe the first few items in the table. Since we encode a count of items, and their size as well, we need not describe each entry in the table.